native/cocos/platform/ios/modules/Battery.mm

```diff
@@ -29,6 +29,8 @@ of this software and associated engine source code (the "Software"), a limited,
 namespace cc {
 
 float Battery::getBatteryLevel() const {
+    // set batteryMonitoringEnabled value to YES otherwise batteryLevel is always -1
+    [UIDevice currentDevice].batteryMonitoringEnabled = YES;
```

Comment:
Is there any problem that set it more than one time?

Reply:
Or should check the value first?

Reply:
No. we don't need